src/renderer/app.tsx

```diff
@@ -1,32 +1,123 @@
+          name="radio"
+          value={radio.value}
+          checked={radioValue === radio.value}
+          onChange={(e) => setRadioValue(e.currentTarget.value)}>
+          {radio.name}
+        </ToggleButton>
+      ))}
+    </ButtonGroup>
+  );
+};
+
+type TweetContProps = {
+  tweets: Tweet[];
+  theme: string;
+};
+
+const TweetContainer = ({ tweets, theme }: TweetContProps) => {
+  const [activeKeyState, setActiveKey] = useState<string>('');
+
+  const handleActiveKeyChange = (activeKey: string) => {
+    if (activeKeyState === activeKey) {
+      //allows us to close expanded item by clicking its toggle while open
+      activeKey = '';
+    }
+    setActiveKey(activeKey);
+  };
+
+  return (
+    <Accordion activeKey={activeKeyState} className={theme}>
+      {tweets.map((tweet) => (
+        <Card key={tweet.id}>
+          <Card.Header>
+            <Accordion.Toggle
+              as={Button}
+              variant="link"
+              eventKey={tweet.id}
+              onClick={() => handleActiveKeyChange(tweet.id)}>
+              {tweet.text.split(' ').slice(0, 15).join(' ')}
+            </Accordion.Toggle>
+          </Card.Header>
+          <Accordion.Collapse eventKey={tweet.id}>
+            <div>
+              <Card.Body>{tweet.text.split(' ').slice(15).join(' ')}</Card.Body>
+              {tweet.videoPath ? <video controls playsInline src={tweet.videoPath} /> : <div></div>}
+            </div>
+          </Accordion.Collapse>
+        </Card>
+      ))}
+    </Accordion>
+  );
+};
+
 const App: FC = () => {
-  const [tweets, setTweets] = useState<Tweet[]>();
-
-  useEffect((): void => {
-    (async () => {
-      let fetchedTweets: Tweet[];
-      do {
-        fetchedTweets = await fetch('https://r-t-generator.herokuapp.com/').then(
-          (response) => response.json() as Promise<Tweet[]>,
-        );
-      } while (!fetchedTweets.length);
-      setTweets(fetchedTweets);
-      // eslint-disable-next-line no-console
-    })().catch(console.error);
-  }, []);
+  const [tweetsList, setTweets] = useState<Tweet[]>();
+  const [radioValue, setRadioValue] = useState('light');
+
+  const updateTweets = (tweets: Tweet[]) => {
+    fetch('https://r-t-generator.herokuapp.com/')
+      .then((response) => response.json())
+      .then((data) => {
+        let fetchedTweets: Tweet[] = data;
+        let i = 0;
+        for (i; i < fetchedTweets.length; i++) {
+          fetchedTweets[i].id = (i + (tweets ? tweets : []).length).toString(10);
+          fetchedTweets[i].videoPath = fetchedTweets[i].videoPath
+            ? 'https://r-t-generator.herokuapp.com/' + fetchedTweets[i].videoPath
+            : undefined;
+        }
+        fetchedTweets = fetchedTweets.concat(tweets ? tweets : []);
+        // console.log(fetchedTweets);
+        setTweets(fetchedTweets);
+      });
+  };
 
+  useEffect(() => {
+    const interval = setInterval((tweets: Tweet[] = tweetsList ? tweetsList : []) => {
```

Comment:
Since you are not passing in this parameter anyway (you are using default values for assignment), why not just let closure do its trick (i.e. remove the parameter, and let the function reference the tweets variable above)